### PR TITLE
Display contact fields (name, surname, company, address, ...) in view component

### PR DESCRIPTION
**Pull request by zatteo**
- Display contact fields (name, surname, company, address, ...) in view component
- Hide by default address subfields and allow to expand them

Screenshot 2024-04-10 at 10 46 17

### CODE REVIEW

File: apps/browser/src/vault/popup/components/vault/view-custom-fields.component.html [hunk 1]
```diff
@@ -3,111 +3,159 @@
...
-        </div>
-        <div *ngIf="field.type === fieldType.Boolean">
-          <i class="bwi bwi-check-square" *ngIf="field.value === 'true'" aria-hidden="true"></i>
-          <i class="bwi bwi-square" *ngIf="field.value !== 'true'" aria-hidden="true"></i>
-          <span class="sr-only">{{ field.value }}</span>
+            *ngIf="field.type != fieldType.Linked"
+            class="row-label draggable"
+            draggable="true"
+            (dragstart)="setTextDataOnDrag($event, field.value)"
+            >{{ field.name }}{{ field.label ? " · " + field.label.type : "" }}</span
```
Comment (Ldoppea) [inline apps/browser/src/vault/popup/components/vault/view-custom-fields.component.html:48]: I'm wondering if the ` · ` character should be rendered in a separate container so we can exclude it from screen readers. As this is a text meant to only provide a graphical separation.

But maybe, by chance, this kind of character is already excluded?

File: apps/browser/src/vault/popup/components/vault/view-custom-fields.component.html [hunk 2]
```diff
@@ -3,111 +3,159 @@
...
-            [field]="field"
-          ></app-vault-view-expiration-date>
-        </div>
-        <div *ngIf="field.type === fieldType.Hidden">
-          <span *ngIf="!field.showValue" class="monospaced">{{ field.maskedValue }}</span>
+    <!--
+      Cozy customization
+      We extracted the cipher.fields for loop in an ng-container and slightly modified below divs condition and style
+    -->
+    <ng-container *ngFor="let field of cipher.fields">
```
Comment (Ldoppea) [inline apps/browser/src/vault/popup/components/vault/view-custom-fields.component.html:27]: (no action needed)
This component starts to be heavily modified. Maybe we are reaching the limits of the `Cozy customization` notation as our modifications are being interlaced. I'm wondering how we can improve this.

**Review by Ldoppea — APPROVED** [review]
Only two nit comments + the suggestion we made during a call about improving the `buildFieldsFromContactByBrowsingModels` methods, then everything is ok to me

File: apps/browser/src/vault/popup/components/vault/view-custom-fields.component.html (hunk 1 above)
Comment (zatteo) [reply apps/browser/src/vault/popup/components/vault/view-custom-fields.component.html:48]: 👍 It seems that screen reader does not have a consistent behavior regarding the `·` character so I will render it in a separate span with `aria-hidden=true`.

File: apps/browser/src/vault/popup/components/vault/view-custom-fields.component.html (hunk 2 above)
Comment (zatteo) [reply apps/browser/src/vault/popup/components/vault/view-custom-fields.component.html:27]: Totally agree. Usual customization comments would have created a mess with the modification I made. That's why I only wrote an explanation comment.

No idea for the moment.

**Comment by zatteo** [comment]
About the _the suggestion we made during a call about improving the buildFieldsFromContactByBrowsingModels methods_ (i.e. a refactoring of buildFieldsFromContactByBrowsingModels to make it simplier), it does not work as is because the `fields` object has incoherences. But this `fields` object may change in the next days with the current work on cozy-contacts. So I prefer to merge it as is now and wait the new version of `fields` to potentially change it.

**Pull request merged**